packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterInputMultipleValue.tsx

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import Autocomplete, { AutocompleteProps } from '@mui/material/Autocomplete';
+import Chip from '@mui/material/Chip';
+
+import TextField from '@mui/material/TextField';
+import { unstable_useId as useId } from '@mui/material/utils';
+import { GridLoadIcon } from '../../icons/index';
+import { GridFilterInputMultipleValueProps } from './GridFilterInputMultipleValueProps';
+
+export const SUBMIT_FILTER_STROKE_TIME = 500;
+
+export interface GridTypeFilterInputMultipleValueProps extends GridFilterInputMultipleValueProps {
+  type?: 'text' | 'number' | 'singleSelect';
+}
+
+function GridFilterInputMultipleValue(
+  props: GridTypeFilterInputMultipleValueProps &
+    Omit<AutocompleteProps<any[], true, false, true>, 'options' | 'renderInput'>,
+) {
+  const { item, applyValue, type, apiRef, ...others } = props;
+  const filterTimeout = React.useRef<any>();
+  const [filterValueState, setFilterValueState] = React.useState(item.value || []);
+  const [applying, setIsApplying] = React.useState(false);
+  const id = useId();
+
+  React.useEffect(() => {
+    return () => {
+      clearTimeout(filterTimeout.current);
+    };
+  }, []);
+
+  React.useEffect(() => {
+    const itemValue = item.value ?? [];
+    setFilterValueState(itemValue.map((x) => String(x)));
+  }, [item.value]);
+
+  const onFilterChange = React.useCallback(
+    (e, value) => {
+      clearTimeout(filterTimeout.current);
+      setFilterValueState(value.map((x) => String(x)));
+
+      if (type !== 'singleSelect' && value === []) {
+        setIsApplying(false);
+        return;
+      }
+
+      setIsApplying(true);
+      // TODO singleSelect doesn't a debounce
+      filterTimeout.current = setTimeout(() => {
+        applyValue({ ...item, value: [...value] });
+        setIsApplying(false);
+      }, SUBMIT_FILTER_STROKE_TIME);
+    },
+    [applyValue, item, type],
+  );
+
+  const InputProps = applying ? { endAdornment: <GridLoadIcon /> } : {};
+
+  return (
+    <Autocomplete
+      {...others}
+      multiple
+      freeSolo
+      options={
+        type === 'singleSelect' ? apiRef.current.getColumn(item.columnField).valueOptions : []
```

Comment:
Instead of an empty array, we could use the text that the user is typing as the first option. It would be closer to https://github.com/mui-org/material-ui-x/issues/2437#issuecomment-903868956. At the first moment, I didn't know I had to press <kbd>Enter</kbd> to filter. BTW, pressing <kbd>Tab</kbd> should also add the typed value as option.